xbmc/addons/AddonManager.cpp

```diff
@@ -934,7 +934,7 @@ bool CAddonMgr::PlatformSupportsAddon(const cp_plugin_info_t *plugin) const
 #if defined(TARGET_FREEBSD)
         || *platform == "freebsd"
 #endif
-        )
+        || *platform == "rbpi")
```

Comment:
That doesn't look correct. It would make it return true for everything not freebsd. Feel free to restructure this whole section because it's really hard to follow..